clients/githubrepo/client.go

```diff
@@ -15,48 +15,89 @@
 package githubrepo
 
 import (
+	"archive/tar"
+	"compress/gzip"
 	"context"
 	"fmt"
 	"io"
 	"io/ioutil"
+	"log"
 	"net/http"
 	"os"
+	"path/filepath"
 	"strings"
 
 	"github.com/google/go-github/v32/github"
+	"github.com/pkg/errors"
 
 	"github.com/ossf/scorecard/clients"
 )
 
-const repoFilename = "githubrepo*.tar.gz"
+const (
+	repoDir      = "repo*"
+	repoFilename = "githubrepo*.tar.gz"
+)
+
+var errZipSlip = errors.New("ZipSlip path detected")
+
+func extractAndValidateArchivePath(path, dest string) (string, error) {
+	const splitLength = 2
+	names := strings.SplitN(path, "/", splitLength)
```

Comment:
so the archive always has a top level directory containing all the actual files? good to leave a comment here to explain this.

Reply:
Yes. Done.